### PR TITLE
MULE-12414: Event Builder allows null message

### PR DESCRIPTION
**Pull request by elrodro83**

### CODE REVIEW

File: core-tests/src/test/java/org/mule/runtime/core/DefaultMuleEventTestCase.java [hunk 1]
```diff
@@ -81,4 +88,16 @@ public void setSessionVariableCustomDataType() throws Exception {
...
   }
+
+  @Test
+  public void setNullMessage() throws Exception {
+    expected.expect(NullPointerException.class);
+    Event.builder(messageContext).message(null);
+  }
+
+  @Test
+  public void dontsetMessage() throws Exception {
```
Comment (dfeist) [inline core-tests/src/test/java/org/mule/runtime/core/DefaultMuleEventTestCase.java]: camel case

File: core-tests/src/test/java/org/mule/runtime/core/DefaultMuleEventTestCase.java [hunk 2]
```diff
@@ -81,4 +88,16 @@ public void setSessionVariableCustomDataType() throws Exception {
     DataType actualDataType = muleEvent.getSession().getPropertyDataType(PROPERTY_NAME);
     assertThat(actualDataType, DataTypeMatcher.like(String.class, APPLICATION_XML, CUSTOM_ENCODING));
   }
+
+  @Test
+  public void setNullMessage() throws Exception {
+    expected.expect(NullPointerException.class);
+    Event.builder(messageContext).message(null);
```
Comment (dfeist) [inline core-tests/src/test/java/org/mule/runtime/core/DefaultMuleEventTestCase.java]: Are we using static imports for builders in the end?

File: core-tests/src/test/java/org/mule/runtime/core/execution/ExceptionToMessagingExceptionExecutionInterceptorTestCase.java [hunk 3]
```diff
@@ -64,6 +65,7 @@ public void before() {
     when(mockMuleContext.getErrorTypeLocator()).thenReturn(mockErrorTypeLocator);
     when(mockMuleEvent.getFlowCallStack()).thenReturn(new DefaultFlowCallStack());
     when(mockMuleEvent.getError()).thenReturn(of(mockError));
+    when(mockMuleEvent.getMessage()).thenReturn(Message.of(null));
```
Comment (dfeist) [inline core-tests/src/test/java/org/mule/runtime/core/execution/ExceptionToMessagingExceptionExecutionInterceptorTestCase.java]: Static?

Comment (elrodro83) [inline core-tests/src/test/java/org/mule/runtime/core/execution/ExceptionToMessagingExceptionExecutionInterceptorTestCase.java]: `of`is already a static import for `Error`

**Pull request merged**